types/mraid/index.d.ts

```diff
@@ -0,0 +1,161 @@
+    audioVolumeChange: (newPercentage: number) => void;
+    adAction: (action: keyof MRAID) => void;
+}
+type MRAIDFeature = "sms" | "tel" | "calendar" | "storePicture" | "inlineVideo" | "vpaid" | "location"; // Does not include supports features from MRAID 1.0 "Candidates for Future Versions" Addendum
+type MRAIDPlacementType = "inline" | "interstitial";
+type MRAIDState = "loading" | "default" | "expanded" | "resized" | "hidden";
+type MRAIDOrientation = "portrait" | "landscape";
+interface MRAIDOrientationProperties {
+    allowOrientationChange: boolean;
+    forceOrientation: MRAIDOrientation | "none";
+}
+interface MRAIDAppOrientationState {
+    orientation: MRAIDOrientation;
+    locked: boolean;
+}
+interface MRAIDPosition {
+    x: number;
+    y: number;
+}
+interface MRAIDSize {
+    width: number;
+    height: number;
+}
+type MRAIDRect = MRAIDPosition & MRAIDSize;
+type MRAIDExpandProperties = MRAIDSize & {
+    // Deprecated in MRAID 3.0; should still be provided by SDKs for backwards compatibility, but may not be honoured
+    useCustomClose: boolean;
+    readonly isModal: boolean;
+};
+type MRAIDResizeProperties = MRAIDSize & {
+    width: number;
+    height: number;
+    allowOffscreen: boolean;
+    // Deprecated in MRAID 3.0; should still be provided by SDKs for backwards compatibility, but may not be honoured
+    customClosePosition?: "top-left" | "top-right" | "bottom-right" | "bottom-left";
+};
+declare enum MRAIDLocationType {
+    LocationServices = 1,
+    IPGeoLocation = 2,
+    UserProvided = 3
+}
+interface MRAIDLocationState {
+    lat: number;
+    lon: number;
+    type: MRAIDLocationType;
+    accuracy: number | undefined;
+    lastfix: number;
+    ipservice: string | undefined;
+}
+interface MRAIDCalendarEvent {
+    id?: string;
+    description: string;
+    location?: string;
+    summary?: string;
+    start: string;
+    end?: string;
+    status?: string;
+    transparency?: string;
+    recurrence?: any;
+    reminder?: string;
+}
+// See VPAID 2.0 spec for further method definitions, but only the declared methods are required by MRAID Video Addendum
+type MRAIDVPAIDObject = {
+    [key: string]: any;
+} & {
+    subscribe(fn: MRAIDVPAIDEventHandlers, event: string, listenerScope?: any): void;
+    unsubscribe(fn: MRAIDVPAIDEventHandlers, event: string): void;
+    startAd(): void;
+    getAdDuration(): number;
+    getAdRemainingTime(): number;
+};
+interface MRAIDVPAIDEventHandlers {
+    AdClickThru: (url: string, id: string, playerHandles: boolean) => void;
+    AdError: (message: string) => void;
+    AdImpression: () => void;
+    AdPaused: () => void;
+    AdPlaying: () => void;
+    AdVideoStart: () => void;
+    AdVideoFirstQuartile: () => void;
+    AdVideoMidpoint: () => void;
+    AdVideoThirdQuartile: () => void;
+    AdVideoComplete: () => void;
+}
+export { MRAID1, MRAID2, MRAID3, MRAIDENVDeclaration,
```

Comment:
Why this big block and not just `export` on each declaration?

Reply:
Initially, I thought of exporting all members at the end of declarations so that one could look at them all in one place, but it makes more sense to do it on each declaration.